tools/linter/adapters/pyrefly_init.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python3
+"""
+Initializer script for pyrefly linter.
+
+This script:
+1. Installs required pip packages for pyrefly
+2. Checks if .pyi stub files exist
+3. If stub files are missing, generates them
+"""
+
+from __future__ import annotations
+
+import argparse
+import subprocess
+import sys
+from pathlib import Path
+
+
+def find_repo_root() -> Path:
+    """Find repository root using git."""
+    try:
+        result = subprocess.run(
+            ["git", "rev-parse", "--show-toplevel"],
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+        return Path(result.stdout.strip())
+    except subprocess.CalledProcessError:
+        sys.exit("Not in a git repository")
+
+
+def install_packages(dry_run: str) -> bool:
+    """Install required pip packages using pip_init.py."""
+    repo_root = find_repo_root()
+    pip_init_script = repo_root / "tools" / "linter" / "adapters" / "pip_init.py"
+
+    packages = [
+        'numpy==2.1.0 ; python_version >= "3.12"',
+        "expecttest==0.3.0",
+        "pyrefly==0.36.2",
+        "sympy==1.13.3",
+        "types-requests==2.27.25",
+        "types-pyyaml==6.0.2",
+        "types-tabulate==0.8.8",
+        "types-protobuf==5.29.1.20250403",
+        "types-setuptools==79.0.0.20250422",
+        "types-jinja2==2.11.9",
+        "types-colorama==0.4.6",
+        "filelock==3.18.0",
+        "junitparser==2.1.1",
+        "rich==14.1.0",
+        "optree==0.17.0",
+        "types-openpyxl==3.1.5.20250919",
+        "types-python-dateutil==2.9.0.20251008",
+    ]
+
+    cmd = [sys.executable, str(pip_init_script), f"--dry-run={dry_run}"] + packages
+
+    print("Installing pyrefly dependencies...")
+    result = subprocess.run(cmd)
+
+    if result.returncode != 0:
+        print("Failed to install packages", file=sys.stderr)
+        return False
+
+    print("Packages installed successfully")
+    return True
+
+
+def check_stub_files() -> tuple[bool, list[str]]:
+    """Check if all required .pyi stub files exist."""
+    repo_root = find_repo_root()
+
+    expected_stub_files = [
+        "torch/_C/__init__.pyi",
+        "torch/_C/_VariableFunctions.pyi",
+        "torch/_C/_nn.pyi",
+        "torch/_VF.pyi",
+        "torch/return_types.pyi",
+        "torch/nn/functional.pyi",
+        "torch/utils/data/datapipes/datapipe.pyi",
+    ]
+
+    missing_files = []
+    for stub_file in expected_stub_files:
+        file_path = repo_root / stub_file
+        if not file_path.exists():
+            missing_files.append(stub_file)
+
+    all_exist = len(missing_files) == 0
+    return all_exist, missing_files
+
+
+def generate_stub_files() -> bool:
```

Comment:
Shouldn't this check occur more than on just init. How can you be sure you aren't checking against stale stub files after updating the repo

Reply:
It's also checking in the `pyrefly_linter`, other than checking for the presence of the files - is there a way to check if they're outdated? That sounds like a great improvement if you know of a mechanism for that :)

Reply:
you could record the last commit hash that you generated the stubs on and if that commit changes regenerate on typecheck.  Although if somebody is updating the stubs this might not cover that case, but I'm assuming that isn't happening often :)

Reply:
@Skylion007 @mlazos attempted to do this in my latest update - please let me know if that was what you're imagining.

Reply:
Yeah this looks good! Thanks